src/util/caseConv.ts

```diff
@@ -53,25 +55,21 @@ function toCamelCase<T>(obj: any, options: ConvertOptions = {}): T {
     return parsed as T;
 }
 
-function snakeCase(str: string): string {
-    let res = '';
-    const isUpper = (c: string) =>
-        'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('').includes(c);
-
-    str.split('').forEach(c => {
-        if (isUpper(c)) res += '_';
-        res += c.toLowerCase();
-    });
+function toSnakeCase<T>(
+    obj: any,
+    options: ConvertOptions = {},
+    visited: Set<any> = new Set<any>(),
```

Comment:
Tracking visited objects isn't necessarily going to work in all cases. Some objects are self-referencing which in turn are also self-referencing something else, so while this could work as a solution, it still risks the same stack recursion limit in #25.

Reply:
You're right! I just pushed a new commit addressing this issue.

In the updated version, we now use a stack to keep track of the objects to process. Instead of making recursive calls, we iterate through the properties of each object in a while loop until the stack is empty. This approach avoids the potential infinite recursion caused by circular references.

Reply:
You're right! I just pushed a new commit addressing this issue.

In the updated version, we now use a stack to keep track of the objects to process. Instead of making recursive calls, we iterate through the properties of each object in a while loop until the stack is empty. This approach avoids the potential infinite recursion caused by circular references.

Please let me know your thoughts!

Reply:
This sounds like a better solution, I'll keep it like this for now and wait for feedback from the affected users.